src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -148,10 +154,23 @@ protected void buildEndPoint(ApiEndpoint baseEndPoint) {
     }
 
     /**
-     * HTTP endpoint to get multiple entities.
+     * HTTP endpoint to get multiple entities based on the user permissions
      */
+    // FIXME Maybe better if the user check (and filtering) was done in a pre hook?
+    // FIXME Will require further granularity for admin
     private List<T> getMany(Request req, Response res) {
-        return persistence.getAll();
+
+        Auth0UserProfile requestingUser = getUserFromRequest(req);
+        if (isUserAdmin(requestingUser)) {
+            return persistence.getAll();
+        } else if (persistence.clazz == OtpUser.class) { 
+            Bson filter = Filters.and(eq("_id", requestingUser.otpUser.id));
```

Comment:
Suggestion: You don't need to use `Filters.and` if there is just one condition. See https://github.com/ibi-group/otp-middleware/blob/4b0cb1ed711fb7ec834563340d2eafb61abba4d0/src/main/java/org/opentripplanner/middleware/auth/Auth0UserProfile.java#L44

Reply:
Updated accordingly